src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/ClaimCreatedOperationHandler.java

```diff
@@ -0,0 +1,170 @@
+package uk.gov.hmcts.cmc.claimstore.events.claim;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.events.ClaimCreationEvent;
+import uk.gov.hmcts.cmc.claimstore.events.operations.ClaimantOperationService;
+import uk.gov.hmcts.cmc.claimstore.events.operations.NotifyStaffOperationService;
+import uk.gov.hmcts.cmc.claimstore.events.operations.RpaOperationService;
+import uk.gov.hmcts.cmc.claimstore.events.operations.UploadOperationService;
+import uk.gov.hmcts.cmc.claimstore.events.solicitor.RepresentedClaimCreatedEvent;
+import uk.gov.hmcts.cmc.claimstore.services.ClaimService;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimState;
+import uk.gov.hmcts.cmc.domain.models.ClaimSubmissionOperationIndicators;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+@Async("threadPoolTaskExecutor")
+@Service
+@ConditionalOnProperty(prefix = "feature_toggles", name = "async_event_operations_enabled")
+public class ClaimCreatedOperationHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ClaimCreatedOperationHandler.class);
+    private final DocumentGenerationService documentGenerationService;
+    private final ClaimService claimService;
+
+    private final Predicate<ClaimSubmissionOperationIndicators> isPinOperationSuccess = indicators ->
+        Stream.of(indicators.getDefendantPinLetterUpload(), indicators.getBulkPrint(),
+            indicators.getStaffNotification(), indicators.getDefendantNotification())
+            .anyMatch(ind -> ind.equals(YesNoOption.NO));
+    private final Predicate<ClaimSubmissionOperationIndicators> isUploadSealedClaimSuccess = indicators ->
+        Stream.of(indicators.getSealedClaimUpload())
```

Comment:
We should not use streams for single enum comparison. Can be replaced with something like:
```
        indicators.getSealedClaimUpload() == YesNoOption.NO;
```
Please change in other places as well.

Reply:
Done